rotkehlchen/chain/evm/utils.py

```diff
@@ -0,0 +1,131 @@
+import logging
+from typing import TYPE_CHECKING, Any, Callable, Final, Optional
+
+from web3.types import BlockIdentifier
+
+from rotkehlchen.assets.asset import EvmToken
+from rotkehlchen.chain.evm.contracts import EvmContract
+from rotkehlchen.constants import ZERO
+from rotkehlchen.constants.resolver import evm_address_to_identifier
+from rotkehlchen.errors.asset import UnknownAsset, WrongAssetType
+from rotkehlchen.errors.misc import BlockchainQueryError, RemoteError
+from rotkehlchen.fval import FVal
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.types import UNISWAP_PROTOCOL, VELODROME_POOL_PROTOCOL, EvmTokenKind, Price
+
+if TYPE_CHECKING:
+    from rotkehlchen.chain.evm.node_inquirer import EvmNodeInquirer
+    from rotkehlchen.types import LP_TOKEN_CONTRACT_ABI_NAMES
+
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+TOKEN_PROTOCOL_TO_ABI_NAME: Final[dict[str, 'LP_TOKEN_CONTRACT_ABI_NAMES']] = {
+    VELODROME_POOL_PROTOCOL: 'VELO_V2_LP',
+    UNISWAP_PROTOCOL: 'UNISWAP_V2_LP',
+}
+MULTICALL_DEPLOYMENT_BLOCK = 7929876
+
+
+def find_price_from_pool_contract(
+        evm_inquirer: 'EvmNodeInquirer',
+        token: EvmToken,
+        token_price_func: Callable,
+        token_price_func_args: list[Any],
+        block_identifier: BlockIdentifier,
+) -> Optional[Price]:
+    """
+    Calculate the price for an LP token the contract of which is also the contract of the
+    pool it represents. For example velodrome lp tokens. The price is calculated like this:
+    price = (Total value of liquidity pool) / (Current supply of LP tokens)
```

Comment:
can you state at the begining of the docstring that this works for any uniswap like pool